docs/reference/migration/migrate_2_0.asciidoc

```diff
@@ -138,6 +138,9 @@ In addition some query builders have been removed or renamed:
 * `filtered(...)` removed. Use `filteredQuery(...)` instead.
 * `inQuery(...)` removed.
 
+The methods IndicesQueryBuilder#parseNoMatchQuery(String) and IndicesQueryBuilder#noMatchQuery(String) have been removed,
```

Comment:
ok I agree with Christoph we could restore the setter that gets a string, but resolve immediately to the proper query builder internally. that would make this change bw compatible as well. we would need to share though this little resolution code between parser and builder. The internal representation should still be only a QueryBuilder